pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1650,6 +1650,21 @@ type S3Bucket struct {
 
 	// UseSDKCreds tells the driver to figure out credentials based on sdk defaults.
 	UseSDKCreds bool `json:"useSDKCreds,omitempty" protobuf:"varint,8,opt,name=useSDKCreds"`
+
+	// CreateBucketIfNotPresent tells the driver to attempt to create the S3 bucket for output artifacts, if it doesn't exist
+	CreateBucketIfNotPresent *CreateS3BucketOptions `json:"createBucketIfNotPresent,omitempty" protobuf:"bytes,9,opt,name=createBucketIfNotPresent"`
+}
+
+// CreateS3BucketOptions options used to determine automatic automatic bucket-creation process
+type CreateS3BucketOptions struct {
+	// Enabled tells the driver whether to attempt to create the S3 bucket or not
+	Enabled bool `json:"enabled,omitempty" protobuf:"varint,1,opt,name=enabled"`
```

Comment:
Do we need `Enabled`? could the existence of `CreateS2BucketOptions` imply enabled?